testplan/runners/pools/remote.py

```diff
@@ -577,19 +579,70 @@ class RemotePoolConfig(PoolConfig):
     Configuration object for
     :py:class:`~testplan.runners.pools.remote.RemotePool` executor
     resource entity.
+    """
+    default_hostname = socket.gethostbyname(socket.gethostname())
+    default_workspace_root = workspace_root()
+
+    @classmethod
+    def get_options(cls):
+        """
+        Schema for options validation and assignment of default values.
+        """
+        return {
+            'hosts': dict,
+            ConfigOption('abort_signals', default=[signal.SIGINT,
```

Comment:
I'm not sure if we need to use ConfigOption at all if we define the default value in the class constructor?

Doesn't need changing for this PR but something to bear in mind in future, I think we can potentially reduce our use of ConfigOption quite a bit.

Reply:
i think this is for differentiating optional and mandatory options

Reply:
We don't need that differentiation when we are unfolding **options and providing a default value in the constructor.